app/models/operatingsystem.rb

```diff
@@ -358,4 +360,16 @@ def reject_empty_provisioning_template(attributes)
     attributes[:_destroy] = 1 if template_exists && provisioning_template_id_empty
     (!template_exists && provisioning_template_id_empty)
   end
+
+  def assign_registration_template
+    template_name = Setting[:default_os_registration_template]
+    template_kind = TemplateKind.unscoped.find_by(name: 'registration')
+    template = ProvisioningTemplate.unscoped.find_by(name: template_name, template_kind: template_kind)
+
+    return unless template
+
+    template.operatingsystems << self
+    template.save
```

Comment:
> Also we should make sure the template is assigned to the org and loc of the OS. Otherwise this could cause problems if e.g. OS is in org A but template only in org B. 

You can assign org/loc to the OS? How?
When I create new org/loc templates are automatically updated.

Reply:
ah sorry for the confusion, what I meant is "OS used in org A", meaning OS has hosts, which are always in some org. So we should make the template available in `ids = os.hosts.distinct.pluck(:organization_id)`. Something like `template.organization_ids = (template.organization_ids + ids).uniq`

Reply:
@ares but in this moment OS has 0 hosts, since its new and this method is called `after_create`, so the `ids` will be always empty array.

Reply:
right, this would only be useful in the migration then